packages/common/src/utils/challenges.ts

```diff
@@ -334,43 +334,68 @@ export const challengeRewardsConfig: Record<
 export const makeOptimisticChallengeSortComparator = (
   userChallenges: Partial<Record<ChallengeRewardID, OptimisticUserChallenge>>
 ): ((id1: ChallengeRewardID, id2: ChallengeRewardID) => number) => {
+  const playCountOrder = [
+    ChallengeName.PlayCount250,
+    ChallengeName.PlayCount1000,
+    ChallengeName.PlayCount10000
+  ]
+
+  const getSortKey = (challenge: OptimisticUserChallenge) => {
```

Comment:
did AI write this? it feels a little more confusing to me bc now i'm comparing the sortKey output and mapping it to -1 or 1.